### PR TITLE
Fix gittuf workflows

### PR DESCRIPTION
**Pull request by adityasaky**

### CODE REVIEW

**Comment by adityasaky** [comment]
https://github.com/in-toto/attestation-verifier/pull/41/files#diff-3df7c2aa640b297b47c907c2ebe29000581edf28349f9d3e8d04aeedfe216d30R35

gittuf changed how ssh keys are used for signing, and this permission must be set on private keys. Basically, gittuf now uses ssh-keygen to sign, and ssh-keygen has this requirement.

**Review by marcelamelara — APPROVED** [review]
LGTM

File: .github/workflows/gittuf-rsl-main.yml [hunk 1]
```diff
@@ -35,8 +31,8 @@ jobs:
           git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
 
           gittuf_key_path="/tmp/gittuf-key"
-
           echo "$KEY" > $gittuf_key_path
+          chmod 600 $gittuf_key_path # ssh-keygen signer requires this
```
Comment (trishankatdatadog) [inline .github/workflows/gittuf-rsl-main.yml:35]: Are we using gitsign (Fulcio) or SSH?

Comment (trishankatdatadog) [reply .github/workflows/gittuf-rsl-main.yml:35]: According to Aditya:

> we're using ssh to sign attestation for the moment, and gitsign to sign the commit itself. once https://github.com/gittuf/gittuf/pull/524 is merged, we can remove the ssh key bits

Comment (marcelamelara) [reply .github/workflows/gittuf-rsl-main.yml:35]: IIUC, the setup-gitsign action is only configuring that every commit is signed, but the key used to sign is the ssh gittuf key.

Comment (marcelamelara) [reply .github/workflows/gittuf-rsl-main.yml:35]: Seeing what [this action](https://github.com/chainguard-dev/actions/blob/main/setup-gitsign/action.yml) does, I am curious as to why we still need lines 27-31.

Comment (trishankatdatadog) [reply .github/workflows/gittuf-rsl-main.yml:35]: Good eye!

Comment (adityasaky) [reply .github/workflows/gittuf-rsl-main.yml:35]: >EDIT: Or I suppose, those may be two distinct signing operations: commit signing which uses gpg vs. RSL signing which uses the ssh key.

this is basically it, except commit signing uses GPG and attestation signing uses SSH. Once https://github.com/gittuf/gittuf/pull/524 is released, we can remove the SSH signing piece.

Comment (adityasaky) [reply .github/workflows/gittuf-rsl-main.yml:29]: >Seeing what this action does, I am curious as to why we still need lines 27-31.

Thanks for catching that, I've removed them now.

**Review by marcelamelara — APPROVED** [review]
Thanks for the updates @adityasaky !

**Pull request merged**